### PR TITLE
Update oracle data documentation

### PR DESCRIPTION
**Pull request by ValentinTrinque**

### CODE REVIEW

File: sources/vega/oracles/v1/data.proto [hunk 1]
```diff
@@ -4,7 +4,8 @@ package oracles.v1;
 
 option go_package = "code.vegaprotocol.io/protos/vega/oracles/v1";
 
-// OracleData describes an oracle data that has been broadcast.
+// OracleData describes a valid oracle data that has been received by the node.
+// It represents both matched an unmatched oracle data.
```
Comment (emilbayes) [inline sources/vega/oracles/v1/data.proto:8]: Typo here in “and”

File: sources/vega/oracles/v1/data.proto [hunk 2]
```diff
@@ -13,9 +14,11 @@ message OracleData {
   // data holds all the properties of the oracle data
   repeated Property data = 2;
   // matched_specs_ids lists all the oracle specs that matched this oracle data.
+  // When it's left empty, it means no oracle spec matched this oracle data.
   repeated string matched_spec_ids = 3;
-  // broadcast_at is the time at which the data was broadcast for the first
-  // time.
+  // broadcast_at is the time at which the data was broadcast to the markets
```
Comment (emilbayes) [inline sources/vega/oracles/v1/data.proto:19]: Perhaps we could be more precise, is time in milliseconds or nanoseconds? (Im guessing the latter based on how we represent time elsewhere)

File: generated/json/data-node/api/v1/ObserveEventBusResponse.json [hunk 3]
```diff
@@ -96,17 +96,17 @@
                         "type": "string"
                     },
                     "type": "array",
-                    "description": "matched_specs_ids lists all the oracle specs that matched this oracle data."
+                    "description": "matched_specs_ids lists all the oracle specs that matched this oracle data. When it's left empty, it means no oracle spec matched this oracle data."
```
[the remark this answers is not in the document]
Comment (ValentinTrinque) [reply generated/json/data-node/api/v1/ObserveEventBusResponse.json:99]: That's an array of the matched IDs. What I meant is "When the array is empty, (obviously) it means nothing has matched". How would you put that ?

Comment (candida-d) [inline generated/json/data-node/api/v1/ObserveEventBusResponse.json:99]:
```suggestion
                    "description": "matched_specs_ids lists all the oracle specs that matched this oracle data. When the array is empty, it means no oracle spec matched this oracle data."
```

[the remark this answers is not in the document]
Comment (candida-d) [reply generated/json/data-node/api/v1/ObserveEventBusResponse.json:99]: I think as you wrote it makes sense! I've added that as a suggestion.

**Review by candida-d — APPROVED** [review]
Once the array comment/suggestion is addressed - looks good to me.

**Pull request merged**